### DISCUSSION TITLE
Using a fixed port for `LocalStackContainer` to use with LocalStack UI

### DISCUSSION THREAD

**Discussion by eyalrin** [testcontainers/testcontainers-java#4794] [Q&A]
Hi,

I’m using Localstack (For SQS for now) via test containers (`LocalStackContainer`). 

Is it possible to set a fixed port (local machine) for it? Each run I get another port (which for normal usage is great) but I want to use the Localstack UI for debugging and it requires me to point to the `host:ip` of my local machine so since the port is changing on each run it forces me to point the UI to a new address every run.

I know there was a `FixedHostPortGenericContainer` but it is both deprecated and doesn't allow me to use the dedicated`LocalStackContainer`...

**Comment by kiview** [comment]
Hey @eyalrin,
I understand your use case, but as you have discovered, this is not supported by Testcontainers. Maybe you can work around this, by logging the Localstack UI URL, so it is more easily accessible for you?

**Comment by eyalrin** [comment]
Hi @kiview,

I don't think it is possible since the Localstack UI is a hosted solution (not something on my machine) - I need to set the URL of my local Localstack/SQS manually in the remote web application.

**Reply by kiview** [reply]
Oh I see, then I don't see a way to easily support this with Testcontainers.
